pkgs/development/compilers/rust/cargo.nix

```diff
@@ -28,6 +28,9 @@ rustPlatform.buildRustPackage rec {
 
   LIBGIT2_SYS_USE_PKG_CONFIG=1;
 
+  # fixes: the cargo feature `edition` requires a nightly version of Cargo, but this is the `stable` channel
```

Comment:
Do they build there own rustc also with rust-nightly?

Reply:
Hum this is a puzzle. They build with stable: https://travis-ci.org/rust-lang/cargo/jobs/428486115#L685-L690
but if I uncomment this line, I get:
```
building
/tmp/nix-build-cargo-1.29.0.drv-0/rustc-1.29.0-src/src/tools/cargo /tmp/nix-build-cargo-1.29.0.drv-0/rustc-1.29.0-src/src
Running cargo build --release
error: failed to parse manifest at `/tmp/nix-build-cargo-1.29.0.drv-0/rustc-1.29.0-src/src/tools/clippy/Cargo.toml`

Caused by:
  the cargo feature `edition` requires a nightly version of Cargo, but this is the `stable` channel
builder for '/nix/store/8z2nqw2x302mcbixsca73d4qazyfq7ys-cargo-1.29.0.drv' failed with exit code 101
```
and indeed in the source tarball the file `rustc-1.29.0-src/src/tools/clippy/Cargo.toml` starts with 
```
cargo-features = ["edition"]
```
and this feature is indeed unstable: https://github.com/rust-lang/cargo/commit/56ee6204de6da133e1c0e5064dfb24065dc41ff7

So I don't know what happens here.